test/libdnf/transaction/test_transaction.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (C) 2017-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "test_transaction.hpp"
+
+#include "libdnf/common/sack/query_cmp.hpp"
+#include "libdnf/transaction/transaction.hpp"
+
+#include <string>
+
+
+using namespace libdnf::transaction;
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION(TransactionTest);
+
+
+static TransactionWeakPtr create_transaction(libdnf::Base & base) {
+    auto trans = base.get_transaction_sack().new_transaction();
+    trans->set_dt_begin(1);
+    trans->set_dt_end(2);
+    trans->set_rpmdb_version_begin("begin");
+    trans->set_rpmdb_version_end("end");
+    trans->set_releasever("26");
+    trans->set_user_id(1000);
+    trans->set_cmdline("dnf install foo");
+
+    trans->add_runtime_package("rpm-4.14.2-1.fc29.x86_64");
```

Comment:
I suggest that this a weak point of implementation. It must contain libdnf::rpm::package, othewise it will be impossible to use it for rpm transaction. Connection with nevra and repoid is not unique.